cvat/apps/engine/models.py

```diff
@@ -912,7 +936,7 @@ def __str__(self):
 
 class Annotation(models.Model):
     id = models.BigAutoField(primary_key=True)
-    job = models.ForeignKey(Job, on_delete=models.CASCADE)
+    job = models.ForeignKey(Job, on_delete=models.DO_NOTHING)
```

Comment:
From [django docs](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.DO_NOTHING), this may leave objects in an inconsistent state. Could you explain why automatic removal on the DB side didn't work for us? Maybe we should just configure `on_delete` manually instead?

Reply:
> Could you explain why automatic removal on the DB side didn't work for us? 

I explained on the meeting. The first point is that will not work with different backends, especially when foreign keys already configured.

Postgres already have setup constraint for ForeignKey fields, and this constraint:

1. Does not allow to remove a row with other rows which refer to this one. So, data will stay consistent. 
2. Goes first, so, we may configure on_delete on level of the database, but it will not work as the first ForeignKey constraint will just abort the operation

Reply:
Try to create a task with a label with an attribute, annotate object and remove LabeledShape without removing before LabeledShapeAttributeVal. You will get an error raised from Postgres.